### PR TITLE
Minimal handling of cancellation

### PR DESCRIPTION
**Pull request by neonichu**
If a command gets marked as cancelled from the llbuild side, we currently do not care at all and treat the build as successful which can be confusing for the user. This at least makes sure clients are notified via the delegate and the build complete message is altered.

rdar://104425290

### CODE REVIEW

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Comment by neonichu** [comment]
Failures look infra related

**Comment by neonichu** [comment]
@swift-ci please smoke test

File: Sources/Build/BuildOperationBuildSystemDelegateHandler.swift [hunk 1]
```diff
@@ -806,8 +821,9 @@ final class BuildOperationBuildSystemDelegateHandler: LLBuildBuildSystemDelegate
         queue.sync {
             self.progressAnimation.complete(success: success)
             if success {
+                let message = cancelled ? "Build cancelled!" : "Build complete!"
```
Comment (abertelrud) [inline Sources/Build/BuildOperationBuildSystemDelegateHandler.swift:824]: Should this say "build failed" if it failed?

Comment (neonichu) [reply Sources/Build/BuildOperationBuildSystemDelegateHandler.swift:824]: Yah, we could do that. I think in most cases it'll be redundant as there's the underlying error right above?

Comment (neonichu) [reply Sources/Build/BuildOperationBuildSystemDelegateHandler.swift:824]: Wait, "Build completed!" should only be shown on success. Cancellation is treated as success (which I agree is confusing), but on failed builds there should be no message at all right now. Are you seeing something different?

Comment (abertelrud) [reply Sources/Build/BuildOperationBuildSystemDelegateHandler.swift:824]: Oh right — you're totally correct.  I was thinking of the `xcodebuild` output.  I'm not seeing anything different, so this looks good to me.  Sorry for the confusion.

**Review by abertelrud — APPROVED** [review]

**Pull request merged**